src/plugins/score-plugin-midi/Midi/MidiDrop.hpp

```diff
@@ -18,11 +18,101 @@ class DropHandler final : public Process::ProcessDropHandler
       const score::DocumentContext& ctx) const noexcept override;
 };
 
+struct PitchbendData{
+  int16_t bend;
+};
+struct ControllerData {
+  midi_size_t channel;
+  midi_size_t number;
+  midi_size_t value;
+  static ControllerData make_cc(const midi_size_t channel, const midi_size_t controller, const midi_size_t value)
+  {
+    SCORE_ASSERT(channel >= 0 && channel < 16);
+    return ControllerData{channel, controller, value};
+  }
+};
+
+struct NoteOnData {
+  midi_size_t channel;
+  midi_size_t note;
+  midi_size_t velocity;
+};
+struct NoteOffData {
+  midi_size_t channel;
+  midi_size_t note;
+  midi_size_t velocity;
+};
+struct MidiTrackEvent {
+  static MidiTrackEvent make_note_off(const double start, const midi_size_t ch, const midi_size_t n, const midi_size_t v){
+    return MidiTrackEvent{m_start: start, m_message:  Midi::NoteOffData{channel: ch, note: n, velocity: v}};
+  }
+  static MidiTrackEvent make_note_on(const double start, const midi_size_t ch, const midi_size_t n, const midi_size_t v){
+    return MidiTrackEvent{m_start: start, m_message:  Midi::NoteOnData{channel: ch, note: n, velocity: v}};
+  }
+  double m_start{};
+
+  void setStart(const double start){ m_start = start; }
+  const double start(){return m_start; }
+  std::variant<Midi::NoteOnData, Midi::NoteOffData, Midi::ControllerData,Midi::PitchbendData> m_message;
+};
+
+struct MidiTrackEvents {
+  void push_back(double delta, int tick, double total, Midi::ControllerData c){
```

Comment:
I wonder if "push_back" is a good name for something that doesn't behave like std::vector::push_back ; it looks closer to emplace_back (but I think that just "add" would be simpler)